packages/flutter/lib/src/rendering/object.dart

```diff
@@ -793,6 +793,37 @@ class PipelineOwner {
       onNeedVisualUpdate();
   }
 
+  /// The unique render object managed by this pipeline that has no parent.
+  RenderObject get rootRenderObject => _rootRenderObject;
+  RenderObject _rootRenderObject;
+  set rootRenderObject(RenderObject value) {
+    if (_rootRenderObject == value)
+      return;
+    _rootRenderObject?.detach();
+    _rootRenderObject = value;
+    _rootRenderObject?.attach(this);
+  }
+
+  /// Calls the given listener whenever the semantics of the render tree change.
+  ///
+  /// Creates [semanticsOwner] if necessary by calling
+  /// [RenderObject.scheduleInitialSemantics]..
+  SemanticsOwner addSemanticsListener(SemanticsListener listener) {
+    if (_semanticsOwner == null) {
+      _semanticsOwner = new SemanticsOwner(onLastListenerRemoved: _handleLastSemanticsListenerRemoved);
```

Comment:
i think i would add the initial listener in the constructor, so that the SemanticsOwner is never in the state of having no listener but not wanting to shut down